Common/Core/CollisionAssociation.h

```diff
@@ -0,0 +1,242 @@
+        // NOTE this does not work anymore if mIncludeUnassigned is set as the unassigned blocks can be somewhere (and we can have merged DFs, too)
+        // if (!iteratorMoved && bcOffset > -bOffsetMax) {
+        //   trackBegin.setCursor(track.filteredIndex());
+        //   iteratorMoved = true;
+        //   LOGP(debug, "Moving iterator begin {}", track.globalIndex());
+        // } else if (bcOffset > bOffsetMax) {
+        //   LOGP(debug, "Stopping iterator {}", track.globalIndex());
+        //   break;
+        // }
+
+        float thresholdTime = 0.;
+        if constexpr (isCentralBarrel) {
+          if (mUsePvAssociation && track.isPVContributor()) {
+            thresholdTime = trackTimeRes;
+          } else if (TESTBIT(track.flags(), track::TrackTimeResIsRange)) {
+            thresholdTime = std::sqrt(sigmaTimeRes2) + mTimeMargin;
+          } else {
+            thresholdTime = mNumSigmaForTimeCompat * std::sqrt(sigmaTimeRes2) + mTimeMargin;
+          }
+        } else {
+          thresholdTime = mNumSigmaForTimeCompat * std::sqrt(sigmaTimeRes2) + mTimeMargin;
+        }
+
+        if (std::abs(deltaTime) < thresholdTime) {
+          const auto collIdx = collision.globalIndex();
+          const auto trackIdx = track.globalIndex();
+          LOGP(debug, "Filling track id {} for coll id {}", trackIdx, collIdx);
+          association(collIdx, trackIdx);
+          if (mFillTableOfCollIdsPerTrack) {
+            if (collsPerTrack[trackIdx] == nullptr) {
+              collsPerTrack[trackIdx] = std::make_unique<std::vector<int>>();
+            }
+            collsPerTrack[trackIdx].get()->push_back(collIdx);
+          }
+        }
+      }
+    }
+    // create reverse index track to collisions if enabled
+    if (mFillTableOfCollIdsPerTrack) {
+      std::vector<int> empty{};
+      for (const auto& track : tracksUnfiltered) {
+
+        const auto trackId = track.globalIndex();
+        if (collsPerTrack[trackId] == nullptr) {
+          reverseIndices(empty);
+        } else {
+          reverseIndices(*collsPerTrack[trackId].get());
+        }
+      }
+    }
+  }
+
+ private:
+  float mNumSigmaForTimeCompat{4.};                      // number of sigma for time compatibility
+  float mTimeMargin{500.};                               // additional time margin in ns
+  int mTrackSelection{TrackSelection::GlobalTrackWoDCA}; // track selection for central barrel tracks (standard association only)
+  bool mUsePvAssociation{true};                          // use the information of PV contributors
+  bool mIncludeUnassigned{true};                         // include tracks that were originally not assigned to any collision
+  bool mFillTableOfCollIdsPerTrack{false};               // fill additional table with vectors of compatible collisions per track
+
+  ClassDefNV(CollisionAssociation, 1);
```

Comment:
Do you intend to serialize objects of this kind into files? If not, there is absolutely no reason to use root stuff here, which will also spare you root dictionary definition. And, more importantly, will solve the issue with expression header being exposed to root cling.

Reply:
Thanks!

Reply:
Hi @aalkin unfortunately this does not solve the compilation issue, do you have any other suggestion I can try?

Reply:
Sorry for the double message, but it should be fixed now with this last commit ([Fix compilation](https://github.com/AliceO2Group/O2Physics/pull/2785/commits/39bbdd01d1edd2c41b9aeff119da94b9ea60d9c1)), so you can ignore the previous one. Thanks!